api/coordinatornetwork/libp2p.go

```diff
@@ -151,7 +124,7 @@ func (coordnet CoordinatorNetwork) announceConnect() error {
 	// Find the other providers for the service CID
 	peerchan := coordnet.dht.FindProvidersAsync(coordnet.ctx, cidvalue, 0)
 	// Connect to peers as they are discovered
-	go handlePeerDiscovery(coordnet.self, peerchan)
+	handlePeerDiscovery(coordnet.self, peerchan)
```

Comment:
The entire codebase (at least AFAIK) tries to delegate the responsibility of using goroutines to the pipeline or higher level components in order to be more flexible and have better control. I believe this is a good practice in general: leave the choice to be asynchronous to the component calling the function instead of doing it inside the function where it could be unexpected